src/main/java/gov/cms/madie/cqllibraryservice/controllers/CqlLibraryController.java

```
@@ -120,6 +123,29 @@ public ResponseEntity<CqlLibrary> updateCqlLibrary(
         .orElseThrow(() -> new ResourceNotFoundException("CQL Library", id));
   }
 
+  @GetMapping(value = "/cql", produces = MediaType.TEXT_PLAIN_VALUE)
+  public String getLibraryCql(
+      @RequestParam String name,
+      @RequestParam String version,
+      @RequestParam Optional<String> model) {
+    List<CqlLibrary> libs =
+        model.isPresent()
+            ? cqlLibraryRepository.findAllByCqlLibraryNameAndDraftAndVersionAndModel(
+                name, false, Version.parse(version), model.get())
+            : cqlLibraryRepository.findAllByCqlLibraryNameAndDraftAndVersion(
```

Just confirming, this will get all Libraries with same name, version, and draft status regardless of model, yes?

Is the expectation that this will be used for searching?

---

Ah, I see we throw an exception if more than 1 library is found. So, we _want_ to filter with model as much as possible.

---

Yeah - this was based on the discussion we had about supporting the different models and what would happen if libraries of two model types had the same version and same name. So model is an optional filter.
I think one of the things we discussed was migrating QDM libraries over and putting them in the same group (using the groupId field) so they can't have the same version as their FHIR counterpart.
But this should at least prevent usage issues by not defaulting to FHIR or QDM if the names and versions overlap.
I'll need to bubble this error condition up in the translator when consuming this API

---

Gotcha. So we want to limit downstream changes until we have a better plan on how QDM libraries will move over to MADiE, hence the optional model parameter and accompanying exception logic.